contest/contest_info.py

```diff
@@ -0,0 +1,156 @@
+'''
+    The MIT License (MIT)
+    Copyright (c) 2014 NTHUOJ team
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+    '''
+from contest.models import Contest
+from contest.models import Contestant
+from contest.models import Clarification
+
+from problem.models import Problem
+from problem.models import Testcase
+from problem.models import Submission
+from problem.models import SubmissionDetail
+
+def get_contestant_list(contest):
+    return Contestant.objects.filter(contest = contest)
+
+def get_contest_submission_list(contest):
+    
+    contestant_list = get_contestant_list(contest)
+    
+    #store contestants' submission dictionary
+    submission_list = []
+    #temporary store single contestant submission
+    contestant_submission_list = []
+    for contestant in contestant_list:
+        for problem in contest.problem.all():
+            submission = Submission.objects.filter(problem = problem , submit_time__lte = contest.end_time,
+                                                   submit_time__gte = contest.start_time , user = contestant.user)
+            contestant_submission_list.append(submission)
+        
+        submission_list.append({'contestant_submission_list':contestant_submission_list})
+        contestant_submission_list = []
+    return submission_list
+
+def get_penalty_scoreboard(contest):
+    
+    contestant_list = get_contestant_list(contest)
+    submission_list = get_contest_submission_list(contest)
+    
+    #store all contestants' data
+    scoreboard = []
+    #store single contestant score
+    contestant_scoreboard = []
+    #store single problem status (pass or not)
+    status = {}
+    #store single problem submit times
+    times = {}
```

Comment:
I suggest to change this variable name to tried_times, and put it into 'for contestant_submission_list in submission_list' loop